src/number_format.js

```diff
@@ -133,6 +133,13 @@ class NumberFormat extends React.Component {
     this.updateValueIfRequired(prevProps);
   }
 
+  componentWillUnmount() {
+    if (this.focusTimeout != null) {
```

Comment:
We don't need this.focusTimeout check. clearTimeout early bails out if we call it with undefined or non-active timeouts.
Same thing we can do it for onBlur.